rdt/transformers/null.py

```diff
@@ -1,87 +1,79 @@
             strategy will be applied depending on whether the input data is numerical
             or not. Defaults to `None`.
-        null_column (bool):
+        model_missing_values (bool):
             Whether to create a new column to indicate which values were null or not.
             If ``None``, only create a new column when the data contains null values.
             If ``True``, always create the new column whether there are null values or not.
             If ``False``, do not create the new column.
             Defaults to ``None``.
-        copy (bool):
-            Whether to create a copy of the input data or modify it destructively.
     """
 
     nulls = None
-    _null_column = None
-    _fill_value = None
+    _model_missing_values = None
+    _missing_value_replacement = None
 
-    def __init__(self, fill_value=None, null_column=None, copy=False):
-        self.fill_value = fill_value
-        self.null_column = null_column
-        self.copy = copy
+    def __init__(self, missing_value_replacement=None, model_missing_values=False):
+        self.missing_value_replacement = missing_value_replacement
+        self.model_missing_values = model_missing_values
+        self._model_missing_values = model_missing_values
 
-    def creates_null_column(self):
+    def creates_model_missing_values(self):
         """Indicate whether this transformer creates a null column on transform.
 
         Returns:
             bool:
                 Whether a null column is created on transform.
         """
-        return bool(self._null_column)
+        return bool(self._model_missing_values)
 
-    def _get_fill_value(self, data, null_values):
+    def _get_missing_value_replacement(self, data):
         """Get the fill value to use for the given data.
 
         Args:
             data (pd.Series):
                 The data that is being transformed.
-            null_values (np.array):
-                Array of boolean values that indicate which values in the
-                input data are nulls.
 
         Return:
             object:
                 The fill value that needs to be used.
         """
-        fill_value = self.fill_value
-
-        if fill_value in (None, 'mean', 'mode') and null_values.all():
-            return 0
+        missing_value_replacement = self.missing_value_replacement
 
-        if fill_value is None:
-            if pd.api.types.is_numeric_dtype(data):
-                fill_value = 'mean'
-            else:
-                fill_value = 'mode'
+        if missing_value_replacement is None:
+            return np.nan
 
-        if fill_value == 'mean':
+        if missing_value_replacement == 'mean':
             return data.mean()
 
-        if fill_value == 'mode':
+        if missing_value_replacement == 'mode':
             return data.mode(dropna=True)[0]
 
-        return fill_value
+        return missing_value_replacement
 
-    def fit(self, data):
+    def fit(self, data, column):
```

Comment:
If the only reason we are passing this is for the warning message then I'm not sure we need it. Couldn't we get the column name by doing `data.name`?